Description: Restores the compatibility with the previous versions
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/junit-platform-runner/src/main/java/org/junit/platform/runner/JUnitPlatform.java
+++ b/junit-platform-runner/src/main/java/org/junit/platform/runner/JUnitPlatform.java
@@ -109,7 +109,6 @@
  * the {@code junit-platform-suite-engine} module; to be removed in JUnit Platform 2.0
  */
 @API(status = DEPRECATED, since = "1.8")
-@Deprecated
 public class JUnitPlatform extends Runner implements Filterable {
 
 	// @formatter:off
--- a/junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java
+++ b/junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java
@@ -96,6 +96,33 @@
 		return testPlan;
 	}
 
+	@API(status = INTERNAL, since = "1.0")
+	public static TestPlan from(Collection<TestDescriptor> engineDescriptors) {
+		return from(engineDescriptors, new ConfigurationParameters() {
+			@Override
+			public Optional<String> get(String key) {
+				return Optional.empty();
+			}
+
+			@Override
+			public Optional<Boolean> getBoolean(String key) {
+				return Optional.empty();
+			}
+
+			@Override
+			@SuppressWarnings("deprecation")
+			public int size() {
+				return 0;
+			}
+
+			@Override
+			public Set<String> keySet() {
+				return java.util.Collections.emptySet();
+			}
+
+		});
+	}
+
 	@API(status = INTERNAL, since = "1.4")
 	protected TestPlan(boolean containsTests, ConfigurationParameters configurationParameters) {
 		this.containsTests = containsTests;
